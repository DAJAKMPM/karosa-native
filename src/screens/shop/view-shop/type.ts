export type ActiveLink = "Popular" | "Latest" | "Price";
