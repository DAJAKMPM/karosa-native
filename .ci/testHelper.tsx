import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
