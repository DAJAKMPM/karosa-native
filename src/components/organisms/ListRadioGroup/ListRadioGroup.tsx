/**
 *
 * ListRadioGroup
 * @format
 *
 */

import React, { FC } from "react";
import { ListItem } from "react-native-elements";
import FormRadioGroup from "@app/molecules/FormRadioGroup";

import type { PropsType } from "./types";
import ListRadioGroupStyles from "./styles";

const ListRadioGroup: FC<PropsType> = (props) => {
  const { name, options } = props;

  return (
    <ListItem>
      <ListItem.Content style={ListRadioGroupStyles.container}>
        <FormRadioGroup name={name} options={options} />
      </ListItem.Content>
    </ListItem>
  );
};

export default ListRadioGroup;
