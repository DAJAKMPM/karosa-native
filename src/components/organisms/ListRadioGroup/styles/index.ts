/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";

const ListRadioGroupStyles = StyleSheet.create({
  container: {
    flexDirection: "column",
  },
});

export default ListRadioGroupStyles;
