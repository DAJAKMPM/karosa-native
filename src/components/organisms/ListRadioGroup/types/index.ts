/**
 *
 * @format
 *
 */

import { Option } from "@app/molecules/FormRadioGroup/types";

export type PropsType = {
  name: string;
  options: Option[];
};
