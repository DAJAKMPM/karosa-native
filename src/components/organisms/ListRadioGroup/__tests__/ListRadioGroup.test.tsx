import React from "react";
import { renderWithFormikWrapper } from "@app/ci/testHelper";
import type { Option } from "@app/components/molecules/FormRadioGroup/types";
import ListRadioGroup from "../ListRadioGroup";

const testData: Option[] = [
  {
    id: "1",
    label: "Lalamove",
    info: "P50",
  },
  {
    id: "2",
    label: "Seller own Courier",
  },
  {
    id: "3",
    label: "Pick Up by Buyer",
  },
];

describe("ListRadioGroup Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithFormikWrapper(
      { test: null },
      <ListRadioGroup name="test" options={testData} />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
