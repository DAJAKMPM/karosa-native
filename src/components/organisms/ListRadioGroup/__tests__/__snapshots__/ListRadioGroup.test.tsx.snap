// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListRadioGroup Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="padView"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "alignItems": "center",
              "flexDirection": "column",
              "justifyContent": "center",
            },
          ]
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <View
          style={
            Object {
              "width": "95%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderBottomWidth": 0.5,
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
              }
            }
            testID="padView"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "space-between",
                  "left": "-5.5%",
                  "position": "relative",
                }
              }
            >
              <View
                accessibilityRole="checkbox"
                accessibilityState={
                  Object {
                    "checked": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                testID="checkbox"
              >
                <View>
                  <SvgMock
                    height={24}
                    width={24}
                  />
                </View>
                <Text
                  style={
                    Object {
                      "color": "#242424",
                      "fontFamily": "SFProText-Regular",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 18.75,
                      "marginLeft": 8,
                    }
                  }
                  testID="checkboxTitle"
                >
                  Lalamove
                </Text>
              </View>
              <View
                style={
                  Object {
                    "left": "70.5%",
                    "position": "relative",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#1C1C1C",
                      "fontFamily": "SFProText-Semibold",
                      "fontSize": 16,
                      "fontWeight": "500",
                      "lineHeight": 18.75,
                    }
                  }
                >
                  P50
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "width": "95%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderBottomWidth": 0.5,
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
              }
            }
            testID="padView"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "space-between",
                  "left": "-5.5%",
                  "position": "relative",
                }
              }
            >
              <View
                accessibilityRole="checkbox"
                accessibilityState={
                  Object {
                    "checked": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                testID="checkbox"
              >
                <View>
                  <SvgMock
                    height={24}
                    width={24}
                  />
                </View>
                <Text
                  style={
                    Object {
                      "color": "#242424",
                      "fontFamily": "SFProText-Regular",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 18.75,
                      "marginLeft": 8,
                    }
                  }
                  testID="checkboxTitle"
                >
                  Seller own Courier
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "width": "95%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderBottomWidth": 0.5,
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
              }
            }
            testID="padView"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "space-between",
                  "left": "-5.5%",
                  "position": "relative",
                }
              }
            >
              <View
                accessibilityRole="checkbox"
                accessibilityState={
                  Object {
                    "checked": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                testID="checkbox"
              >
                <View>
                  <SvgMock
                    height={24}
                    width={24}
                  />
                </View>
                <Text
                  style={
                    Object {
                      "color": "#242424",
                      "fontFamily": "SFProText-Regular",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 18.75,
                      "marginLeft": 8,
                    }
                  }
                  testID="checkboxTitle"
                >
                  Pick Up by Buyer
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
